### PR TITLE
Service discovery

### PR DESCRIPTION
**Pull request by maartenvanderhoef**
## What

Adding service discovery for ECS Services.

## Why

We should have it.

### CODE REVIEW

**Comment by maartenvanderhoef** [comment]
@Jamie-BitFlight I more or less copied something you did earlier, with merging service_discovery_properties_defaults with service_discovery_properties so we don't have large sets of default string vars anymore. Let me know what you think about it.

File: main.tf [hunk 1]
```diff
@@ -301,6 +310,28 @@ module "ecs_service" {
...
+  service_discovery_dns_type = "${lookup(local.service_discovery_properties,"dns_type")}"
+
+  # routing_policy def
+  service_discovery_routing_policy = "${lookup(local.service_discovery_properties,"routing_policy")}"
+
+  # healthcheck_custom_failure_threshold needed, set to 1 by default
+  service_discovery_healthcheck_custom_failure_threshold = "${lookup(local.service_discovery_properties,"healthcheck_custom_failure_threshold")}"
+
+  # tags
+  tags = "${var.tags}"
```
Comment (Jamie-BitFlight) [inline main.tf]: local.tags

File: variables.tf [hunk 2]
```diff
@@ -423,3 +439,44 @@ variable "ecs_cron_tasks" {
...
+  default = "false"
+}
+
+# The service discovery namespace id to register the services with
+variable "service_discovery_namespace_id" {
+  default = ""
+}
+
+## Defaults for the service_discovery_properties
+variable "service_discovery_properties_defaults" {
```
Comment (Jamie-BitFlight) [inline variables.tf]: like this :)

Comment (maartenvanderhoef) [reply variables.tf]: Yeah me too, we should apply this pattern everywhere! :-).

**Pull request merged**